### PR TITLE
Functional component example

### LINKED ISSUE #23: React functional component example
### What should we do?

Make a new example that uses React Functional component. Optional with Typescript.

### Why should we do it?

React is moving away from Class based component. I already have a working example.  https://github.com/joey-kwl/React-Typescript-BPMN-example

### PR DESCRIPTION
**Pull request by joey-kwl**
Updated example with React functional component. It also uses React 18

Closes #23

### CODE REVIEW

File: example/public/index.html [hunk 1]
```diff
@@ -53,8 +53,7 @@ <h1>Display BPMN 2.0 Diagrams Using React</h1>
       Suspendisse potenti. In ut lorem vitae dolor molestie commodo. Vivamus mattis gravida metus, porttitor ornare augue lacinia suscipit. Duis condimentum fringilla nisl, non consequat odio mollis id. Nulla tincidunt interdum elit, sed tristique ex laoreet lobortis. Suspendisse in felis vitae magna convallis semper. Nullam odio risus, venenatis eget erat ac, pretium aliquet urna. Duis luctus lobortis maximus.
     </p>
   </div>
-  <script type="text/babel">
-
+    <script type="text/babel">
```
Comment (nikku) [inline example/public/index.html]:
```suggestion
  <script type="text/babel">
```

File: example/public/index.html [hunk 2]
```diff
@@ -67,65 +66,63 @@ <h1>Display BPMN 2.0 Diagrams Using React</h1>
...
-                onLoading={ onLoading }
-                onShown={ onShown }
-                onError={ onError }
-              />
-            : null
-        );
-      }
-
+    function ExternalLoadedBpmn({ url }) {
+      const [diagramXML, setDiagramXML] = React.useState()
```
Comment (nikku) [inline example/public/index.html]:
```suggestion
      const [diagramXML, setDiagramXML] = React.useState();
```

File: example/public/index.html [hunk 3]
```diff
@@ -67,65 +66,63 @@ <h1>Display BPMN 2.0 Diagrams Using React</h1>
...
+            onShown={onShown}
+            onError={onError}
+          />
+          : null
+      );
     }
 
-    ReactDOM.render(
+    const container = document.querySelector('#diagram');
+    const root = ReactDOM.createRoot(container)
```
Comment (nikku) [inline example/public/index.html]: Is `ReactDOM.render` reprecated? If not I'd rather use it as it si the post simple version.

**Review by nikku — COMMENTED** [review]
Thanks for your addition. Could you add missing `;` behind individual code statements? This is our style guide; we somehow are just not able to validate it here.

File: example/public/index.html (hunk 3 above)
Comment (joey-kwl) [reply example/public/index.html]: `ReactDOM.render` is deprecated in React 18: https://react.dev/blog/2022/03/29/react-v18#react-dom-client

**Comment by joey-kwl** [comment]
> Thanks for your addition. Could you add missing `;` behind individual code statements? This is our style guide; we somehow are just not able to validate it here.

Will do it

File: example/public/index.html (hunk 3 above)
Comment (nikku) [reply example/public/index.html]: Thanks, did not know! :)

**Review by nikku — APPROVED** [review]
Looks great!

**Pull request merged**